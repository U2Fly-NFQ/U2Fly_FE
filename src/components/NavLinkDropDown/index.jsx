import { DownOutlined } from '@ant-design/icons'
import React from 'react'
import NavLinkCustom from '../NavLink'
import './index.scss'
export default function NavLinkDropDown({ Title, ListDropDown }) {
  return (
    <li className="navbar__link">
      <NavLinkCustom to="/News">
        <p className="navbar__link__data">
          {Title} <DownOutlined />{' '}
        </p>
      </NavLinkCustom>
      {ListDropDown.length > 0 && (
        <div className="navbar__dropdown">
          <div className="navbar__dropdown__content">
            <div className="navbar__dropdown__lists">
              <ul className="list__dropdown__items">
                {ListDropDown.map((item) => {
                  return <li className="list__dropdown__item">{item}</li>
                })}
              </ul>
            </div>
          </div>
        </div>
      )}
    </li>
  )
}
