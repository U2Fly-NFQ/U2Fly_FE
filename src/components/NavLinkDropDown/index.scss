@use '../../assets/scss/abstracts/' as b;
.navbar__link {
  padding: 0 18px;
  font-weight: 600;
  font-size: 14px;
  cursor: pointer;
  transition: all 0.3s ease;
  display: table-cell;
  position: relative;
  list-style-type: none;
  a {
    text-decoration: none;
  }

  .navbar__dropdown {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    font-weight: 500;
    float: left;
    min-width: 10rem;
    font-size: 1rem;
    transform: translateY(30px);
    opacity: 0;
    color: #212529;
    text-align: left;
    visibility: hidden;
    list-style: none;
    background-color: #fff;
    background-clip: padding-box;

    border-radius: 0.25rem;
    transition: all 0.5s ease;
    .navbar__dropdown__content {
      display: flex;
      gap: 3rem;
      margin-bottom: 8px;

      .navbar__dropdown__lists {
        flex: 1;
        .title {
          position: relative;
        }
        .title::after {
          border-bottom: 1px solid #cea384;
          bottom: 0;
          content: '';
          position: absolute;
          transition: width 0.4s linear 0s;
          left: 0;
          width: 50%;
        }
        .list__dropdown__items {
          padding: 0;
          margin-top: 3px;
          list-style-type: none;
          display: flex;
          flex-direction: column;
          gap: 8px;
          .list__dropdown__item {
            text-decoration: none;
            font-size: 15px !important;
            color: #7d7b7b;
            flex: 1;
            padding: 4px 15px;
            line-height: 28px;
            text-transform: unset;
            border-radius: 2px;
            letter-spacing: unset;
            transition: all 0.5s ease;
          }
          .list__dropdown__item:hover {
            color: b.$main-color;
            background-color: #f7f7f7;
          }
        }
      }
    }

    .navbar__video {
      position: relative;
      overflow: hidden;
      max-width: 100%;
      padding-bottom: 56.25%;
      height: auto;
      iframe {
        position: absolute;
        margin-top: 8px;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        width: 100%;
        height: 100%;
      }
    }
  }
}
