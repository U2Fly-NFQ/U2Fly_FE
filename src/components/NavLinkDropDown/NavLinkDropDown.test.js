import renderer from 'react-test-renderer'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import NavLinkDropDown from './'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../translations'
describe('Navigation bar test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <Router>
            <NavLinkDropDown Title={{ title: 'home', path: '/hello' }} />
          </Router>
        </I18nextProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
