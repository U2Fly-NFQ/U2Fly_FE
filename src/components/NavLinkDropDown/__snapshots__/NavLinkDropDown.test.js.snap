// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar test renders correctly 1`] = `
<li
  className="navbar__link"
>
  <a
    className="nav-link"
    href="//hello"
    onClick={[Function]}
  >
    <p
      className="navbar__link__data"
    >
      home
       
    </p>
  </a>
</li>
`;
